plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java

```diff
@@ -1018,6 +1018,25 @@ a_enum_2 Enum('hello', 'world', 'a', 'b', 'c'))
             assertThat(query(smallDomainCompactionThreshold, "SELECT some_column FROM " + table.getName() + " WHERE a_string NOT IN ('a', 'b')" + withConnectorExpression)).isNotFullyPushedDown(FilterNode.class);
+
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string LIKE 'b'")).isFullyPushedDown();
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string LIKE 'b'" + withConnectorExpression)).isFullyPushedDown();
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string LIKE 'b%'")).isFullyPushedDown();
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string LIKE 'b%'" + withConnectorExpression)).isFullyPushedDown();
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string LIKE '%b'")).isFullyPushedDown();
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string LIKE '%b'" + withConnectorExpression)).isFullyPushedDown();
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string LIKE '%b%'")).isFullyPushedDown();
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string LIKE '%b%'" + withConnectorExpression)).isFullyPushedDown();
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_enum_1 LIKE '%b%'")).isNotFullyPushedDown(FilterNode.class);
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_enum_1 LIKE '%b%'" + withConnectorExpression)).isNotFullyPushedDown(FilterNode.class);
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string LIKE a_string_alias")).isFullyPushedDown();
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string LIKE a_string_alias" + withConnectorExpression)).isFullyPushedDown();
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string LIKE a_enum_1")).isNotFullyPushedDown(FilterNode.class);
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string LIKE a_enum_1" + withConnectorExpression)).isNotFullyPushedDown(FilterNode.class);
+            assertThat(query(convertToVarchar, "SELECT some_column FROM " + table.getName() + " WHERE a_string LIKE unsupported_1")).isNotFullyPushedDown(FilterNode.class);
+            assertThat(query(convertToVarchar, "SELECT some_column FROM " + table.getName() + " WHERE a_string LIKE unsupported_1" + withConnectorExpression)).isNotFullyPushedDown(FilterNode.class);
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string NOT LIKE '%b%'")).isFullyPushedDown();
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string NOT LIKE '%b%'" + withConnectorExpression)).isFullyPushedDown();
```

Comment:
Additional test case for `LIKE` with escape and `a_string` being an enum

Reply:
Clickhouse does not support like with escape.

If eve usual like is not supported for enum, then all the more negation will not be supported.
```
            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_enum_1 LIKE '%b%'")).isNotFullyPushedDown(FilterNode.class);
            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_enum_1 LIKE '%b%'" + withConnectorExpression)).isNotFullyPushedDown(FilterNode.class);
```
However I'll provide this additional case to be sure about that.

Reply:
The additional test case is to ensure that our pushdown rule doesn't capture `LIKE` with `ESCAPE` so that we would apply the filter in the Trino side